iceoryx_posh/source/runtime/posh_runtime.cpp

```diff
@@ -87,11 +87,28 @@ PoshRuntime::PoshRuntime(cxx::optional<const ProcessName_t*> name, const bool do
     {
         LogWarn() << "Running applications on 32-bit architectures is not supported! Use at your own risk!";
     }
+
+    /// @todo #404 Use posix::FileLock here
+    //
+    // if (m_fileLock.hasError())
+    // {
+    //     // If the lock couldn't be acquired, some application with the same name is still running
+    //     LogError() << "An application with the same name is still running. Starting the same app twice is not
+    //                   supported.";
+    //                   errorHandler(
+    //                       Error::kPOSH__RUNTIME_APP_WITH_SAME_RUNTIME_NAME_STILL_RUNNING, nullptr, iox::ErrorLevel::FATAL);
+    // }
+
     /// @todo here we could get the LogLevel and LogMode and set it on the LogManager
 }
 
 PoshRuntime::~PoshRuntime() noexcept
 {
+    // Inform RouDi that we're shuting down
+    IpcMessage shutdown;
+    shutdown << IpcMessageTypeToString(IpcMessageType::TERMINATION) << m_appName;
+    m_ipcChannelInterface.sendMessageToRouDi(shutdown);
+
     if (m_applicationPort)
     {
         m_applicationPort.destroy();
```

Comment:
This is a race. RouDi already cleans up the application port and might have already delivered it to another application. `sendMessageToRouDi` is also not used anywhere else. Please consider using `sendRequestToRoudi`, then let RouDi respond with an `ACK` and cleanup the ports. The runtime wouldn't do anything.
`sendMessageToRouDi` should probably be removed.

Reply:
You're right, done. I hope we can solve this even cleaner with inter-process RAII in #611